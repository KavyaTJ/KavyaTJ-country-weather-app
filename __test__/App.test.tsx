import { act, render } from '@testing-library/react';
import App from '../App';
import {
    componentRenderByMemoryRouter,
    toBeExpectByTestId,
    toBeExpectByText,
} from '../types/testUtils';

describe('Test App Router', () => {
    test('should render app componet', () => {
        componentRenderByMemoryRouter('/', <App />);
        toBeExpectByTestId('app-component-test-id');
    });
    test("snapshot testing", () => {
        const InputProps = {
          onChange: jest.fn(),
          className: "inputField",
          placeHolder: "Enter the country name",
        };
        const { container } = render(<input {...InputProps} />);
        expect(container.firstChild).toMatchSnapshot();
      });
    test('should render 404 page', () => {
        componentRenderByMemoryRouter(
            '/details/BD/hjgsdsdjfg',
            <App />
        );
        toBeExpectByText('404 Page Not Found');
    });
});
