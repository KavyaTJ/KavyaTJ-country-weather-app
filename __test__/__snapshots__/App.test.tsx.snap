// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Router snapshot testing 1`] = `
<input
  class="inputField"
  placeholder="Enter the country name"
/>
`;
